integration-tests/smoke/ccip/ccip_reorg_test.go

```diff
@@ -0,0 +1,675 @@
+	destChainSelector := allChains[1]
+
+	dockerEnv, ok := tEnv.(*testsetups.DeployedLocalDevEnvironment)
+	require.True(t, ok)
+
+	chainSelToRPCURL := make(map[uint64]string)
+	for _, chain := range dockerEnv.GetDevEnvConfig().Chains {
+		require.GreaterOrEqual(t, len(chain.HTTPRPCs), 1)
+		details, err := chainsel.GetChainDetailsByChainIDAndFamily(strconv.FormatUint(chain.ChainID, 10), chainsel.FamilyEVM)
+		require.NoError(t, err)
+
+		chainSelToRPCURL[details.ChainSelector] = chain.HTTPRPCs[0].Internal
+	}
+
+	reorgSourceClient := ctf_client.NewRPCClient(chainSelToRPCURL[reorgSourceSelector], nil)
+
+	testhelpers.AddLaneWithDefaultPricesAndFeeQuoterConfig(t, &e, state, reorgSourceSelector, destChainSelector, false)
+
+	// wait for log poller filters to get registered.
+	l.Info().Msg("waiting for log poller filters to get registered")
+	time.Sleep(15 * time.Second)
+	reorgingMsgEvent := testhelpers.TestSendRequest(t, e.Env, state, reorgSourceSelector, destChainSelector, false, router.ClientEVM2AnyMessage{
+		Receiver:     common.LeftPadBytes(state.Chains[destChainSelector].Receiver.Address().Bytes(), 32),
+		Data:         []byte("hello world"),
+		TokenAmounts: nil,
+		FeeToken:     common.HexToAddress("0x0"),
+		ExtraArgs:    nil,
+	})
+	l.
+		Info().
+		Str("messageID", hexutil.Encode(reorgingMsgEvent.Message.Header.MessageId[:])).
+		Uint64("messageBlockNumber", reorgingMsgEvent.Raw.BlockNumber).
+		Str("messageBlockHashBeforeReorg", reorgingMsgEvent.Raw.BlockHash.String()).
+		Msg("sent CCIP message that will get re-orged before getting finalized")
+
+	const reorgDepth = 7
+	var minChainBlockNumberBeforeReorg = reorgingMsgEvent.Raw.BlockNumber + reorgDepth - 1
+	// let reorgDepth - 1 blocks pass by before re-orging the message.
+	// This will effectively rewind the chain to a block where the message didn't exist.
+	require.Eventually(t, func() bool {
+		bn, err := reorgSourceClient.BlockNumber()
+		require.NoError(t, err)
+		l.Info().
+			Int64("blockNumber", bn).
+			Uint64("targetBlockNumber", minChainBlockNumberBeforeReorg).
+			Msg("Waiting for chain to progress above target block number")
+		return bn >= int64(minChainBlockNumberBeforeReorg) //nolint:gosec // this will not overflow in the current test
+	}, 1*time.Minute, 500*time.Millisecond, "timeout exceeded: chain did not progress above the target block number")
+
+	// Run reorg below finality depth
+	l.Info().
+		Uint64("messageBlockNumber", reorgingMsgEvent.Raw.BlockNumber).
+		Int("reorgDepth", reorgDepth).
+		Uint64("sourceChainSelector", reorgSourceSelector).
+		Msg("starting blockchain reorg on Simulated Geth chain")
+	err = reorgSourceClient.GethSetHead(reorgDepth)
+	require.NoError(t, err, "error starting blockchain reorg on Simulated Geth chain")
+
+	bnAfterReorg, err := reorgSourceClient.BlockNumber()
+	require.NoError(t, err, "error getting block number after reorg")
+
+	l.Info().Int64("blockNumberAfterReorg", bnAfterReorg).Msg("block number after reorg")
```

Comment:
how about a check to validate whether reorg happened? bnAfterReorg < minChainBlockNumberBeforeReorg